src/main/java/cubicchunks/worldgen/GeneratorStage.java

```diff
@@ -23,27 +23,63 @@
  */
 package cubicchunks.worldgen;
 
-public enum GeneratorStage {
-	TERRAIN,
-	SURFACE,
-	FEATURES,
-	LIGHTING,
-	POPULATION,
-	LIVE;
-
-	public static GeneratorStage getFirstStage() {
-		return values()[0];
+public class GeneratorStage {
+	
+	public static GeneratorStage LIGHTING = new GeneratorStage("lighting");
+	
+	public static GeneratorStage LIVE = new GeneratorStage("live");
+	static {
+		LIVE.setLastStage();
+		LIVE.setOrdinal(Integer.MAX_VALUE);
 	}
-
-	public static GeneratorStage getLastStage() {
-		return values()[values().length - 1];
+	
+	private final String name;
+	
+	private boolean isLast;
+	
+	private int ordinal;
+	
+	private StageProcessor processor;
+	
+		
+	public GeneratorStage(String name) {
+		this.name = name;
+		this.isLast = false;
+	}
+	
+	
+	public String getName() {
+		return this.name;
+	}
+	
+	
+	public void setLastStage() {
+		this.isLast = true;
 	}
 
 	public boolean isLastStage() {
-		return ordinal() == values().length - 1;
+		return this.isLast;
+	}
+	
+	
+	public void setProcessor(StageProcessor processor) {
+		this.processor = processor;
+	}
+
+	public StageProcessor getProcessor() {
+		return this.processor;
+	}
+	
+	
+	public void setOrdinal(int ordinal) {
```

Comment:
Ordinals are the fastest way of determining if a given stage precedes a different stage. If I were to remove them, isPreceding would require iterating over the stage list. Ordinals are being automatically assigned when they are being added to the pipeline. I'm not certain if a builder would solve this significantly better to justify the added complexity. I'll set the setter and getter to package private for now.